File: Python/Catalan.py
def CatalanNumber(n):
    dpVec = [0 for i in range(n + 1)]
    dpVec[0] = 1
    for i in range(1, n + 1):
        for j in range(0, i):
            dpVec[i] += dpVec[i - 1 - j] * dpVec[j]
    return dpVec[n]


test = int(input())
for i in range(test):
    print(CatalanNumber(int(input())))

File: C++/Catalan.cpp
#include <iostream>
#include <vector>
using namespace std;
long long int catalanNumber(int n)
{
    vector<long long int> dpVec(n + 1, 0);
    dpVec[0] = 1;
    for (int i = 1; i <= n; i++)
    {
        for (int j = 0; j < i; j++)
        {
            dpVec[i] += dpVec[i - 1 - j] * dpVec[j];
        }
    }
    return dpVec[n];
}
int main()
{
    int test;
    cin >> test;
    while (test--)
    {
        long long int batao;
        cin >> batao;
        cout << catalanNumber(batao) << endl;
    }
}